service/accounts/openstack_manager.py

```diff
@@ -485,33 +435,124 @@ def delete_security_group(self, identity):
             self.network_manager.neutron.delete_security_group(sec_group["id"])
         return True
 
-    def delete_network(self, identity, remove_network=True):
-        # Core credentials need to be converted to openstack names
+    def select_network_strategy(self, identity, topology_name=None):
+        """
+        Select a network topology and initialize it with the identity/provider specific information required.
+        """
+        # Select Cls
+        if not topology_name:
+            NetworkTopologyStrategyCls = ExternalRouter
+        else:
+            NetworkTopologyStrategyCls = get_topology_cls(topology_name)
+
+        try:
+            network_strategy = NetworkTopologyStrategyCls(identity)
+        except:
+            raise
+            raise Exception(
+                "Error initializing Network Topology - %s + %s " %
+                (NetworkTopologyStrategyCls, identity))
```

Comment:
The second raise will never be raised. Convert the second exception to a print statement then simple `raise` after?